File: frontend/src/api/auditLogs.ts
import apiClient from './client'

export interface AuditLog {
  id: string
  user_id: string
  action: string
  resource_type: string
  resource_id: string | null
  details: string | null
  ip_address: string | null
  created_at: string
}

export interface AuditLogListResponse {
  items: AuditLog[]
  total: number
  page: number
  page_size: number
}

export interface AuditLogParams {
  action?: string
  user_id?: string
  start_date?: string
  end_date?: string
  page?: number
  page_size?: number
}

export const auditLogsApi = {
  list: (params?: AuditLogParams) =>
    apiClient.get<AuditLogListResponse>('/admin/audit-logs', { params }),
}

File: backend/meks/api/schemas/audit_logs.py
from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: str
    user_id: str | None
    action: str
    resource_type: str
    resource_id: str | None
    details: str | None
    ip_address: str | None
    user_agent: str | None
    created_at: str

    model_config = {"from_attributes": True}


class AuditLogListResponse(BaseModel):
    items: list[AuditLogResponse]
    total: int
    page: int
    page_size: int
